src/Core/Framework/Framework.php

```diff
@@ -166,27 +167,39 @@ private function registerEntityExtensions(
         ExtensionRegistry $registry
     ): void {
         foreach ($registry->getExtensions() as $extension) {
-            /** @var string $class */
-            $class = $extension->getDefinitionClass();
-
-            try {
-                $definition = $definitionRegistry->get($class);
-            } catch (DefinitionNotFoundException) {
-                continue;
-            }
-
-            $definition->addExtension($extension);
-
-            try {
-                $salesChannelDefinition = $salesChannelRegistry->get($class);
-            } catch (DefinitionNotFoundException) {
-                continue;
-            }
-
-            // same definition? do not added extension
-            if ($salesChannelDefinition !== $definition) {
-                $salesChannelDefinition->addExtension($extension);
-            }
+            $this->addExtension($definitionRegistry, $salesChannelRegistry, $extension);
+        }
+
+        foreach ($registry->buildBulkExtensions($definitionRegistry) as $extension) {
+            $this->addExtension($definitionRegistry, $salesChannelRegistry, $extension);
+        }
+    }
+
+    private function addExtension(
+        DefinitionInstanceRegistry $definitionRegistry,
+        SalesChannelDefinitionInstanceRegistry $salesChannelRegistry,
+        EntityExtension $extension
+    ): void {
+        /** @var string $class */
```

Comment:
```suggestion
```

Type is already present as return type: https://github.com/shopware/shopware/blob/trunk/src/Core/Framework/DataAbstractionLayer/EntityExtension.php#L34